extensions/blocks/podcast-player/podcast-player.js

```diff
@@ -0,0 +1,372 @@
+		if ( ! this.tracks.length ) {
+			return;
+		}
+
+		// Initialize player UI.
+		this.initializeAudio();
+
+		// Store track elements and cast as Array.
+		this.trackElements = [ ...block.querySelector( '.podcast-player__episodes' ).children ];
+
+		// Attach event handlers.
+		this.block.addEventListener( 'click', this.handleClick.bind( this ), false );
+		this.block.addEventListener( 'keydown', this.handleKeyDown.bind( this ), false );
+	}
+
+	/**
+	 * Lookup JSON data in block and load them into this intance.
+	 * @private
+	 */
+	loadMetadata() {
+		const dataContainer = this.block.querySelector( 'script[type="application/json"]' );
+		if ( dataContainer ) {
+			try {
+				const data = JSON.parse( dataContainer.text );
+				this.tracks = data.tracks;
+				this.attributes = data.attributes;
+			} catch ( e ) {
+				return;
+			}
+		}
+	}
+
+	/**
+	 * Initialize audio player instance.
+	 * @private
+	 */
+	initializeAudio() {
+		// Construct audio element.
+		this.audio = document.createElement( 'audio' );
+		this.audio.src = this.getTrack( 0 ).src;
+		this.audio.addEventListener( 'play', () => this.handlePlay() );
+		this.audio.addEventListener( 'pause', () => this.handlePause() );
+		this.audio.addEventListener( 'error', () => this.handleError() );
+
+		// Insert player into the DOM.
+		this.block.insertBefore( this.audio, this.block.firstChild );
+
+		// Initialize MediaElement.js
+		this.mediaElement = new MediaElementPlayer( this.audio, meJsSettings );
+	}
+
+	/**
+	 * Get track data.
+	 * @private
+	 * @param {number} track The track number
+	 * @returns {object} Track object.
+	 */
+	getTrack( track ) {
+		return this.tracks[ track ];
+	}
+
+	/**
+	 * Error handler for audio.
+	 * @private
+	 */
+	handleError() {
+		this.setPlayerState( STATE_ERROR );
+		if ( this.currentTrack > -1 ) {
+			this.setTrackState( this.currentTrack, true, true );
+		}
+	}
+
+	/**
+	 * Play handler for audio.
+	 * @private
+	 */
+	handlePlay() {
+		this.setPlayerState( STATE_PLAYING );
+		if ( this.currentTrack === -1 ) {
+			this.currentTrack = 0;
+			this.setTrackState( this.currentTrack, true );
+		}
+	}
+
+	/**
+	 * Pause handler for audio.
+	 * @private
+	 */
+	handlePause() {
+		// Ignore pauses if we are showing an error.
+		if ( this.playerState === STATE_ERROR ) {
+			return;
+		}
+
+		this.setPlayerState( STATE_PAUSED );
+	}
+
+	/**
+	 * Play current audio.
+	 * @public
+	 */
+	play() {
+		// Ignoring exceptions as they are handled globally from the audio element.
+		this.audio.play().catch( noop );
+	}
+
+	/**
+	 * Pause current audio.
+	 * @public
+	 */
+	pause() {
+		this.audio.pause();
+	}
+
+	/**
+	 * Stop current audio, clean selected track.
+	 * @public
+	 */
+	stop() {
+		this.pause();
+		this.setTrackState( this.currentTrack, false );
+		this.currentTrack = -1;
+	}
+
+	/**
+	 * Toggle playing state.
+	 * @public
+	 */
+	togglePlayPause() {
+		if ( this.audio.paused ) {
+			this.play();
+		} else {
+			this.pause();
+		}
```

Comment:
Is it preferred to do an expanded state of this for readability instead of `this.audio.paused ? this.play() : this.pause()`? I'm really just asking here so I learn the conventions here :)

Reply:
I'm not 100% sure there is a convention but I'm personally avoiding ternary operators to just call functions. I only use them where I care about the value it returns.